src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../textbox';
+import I18nHelper from '../../../utils/helpers/i18n';
+import Logger from '../../../utils/logger';
+
+class Decimal extends React.Component {
+  input = null;
+
+  // Create ref to document for tests
+  _document = document;
+
+  formatValue = () => {
+    const { value } = this.props;
+    const { input } = this;
+
+    // Return unformatted value if component has not mounted
+    if (!input) {
+      return value;
+    }
+    // Return unformatted value if input is being used
+    if (input && this._document.activeElement === input.current) {
+      return value;
+    }
+
+    // Only format value if input is not active
+    // Strip delimiters otherwise formatDecimal Helper goes nuts
+    const format = I18nHelper.format();
+    const delimiter = `\\${format.delimiter}`;
+    const delimiterMatcher = new RegExp(`[${delimiter}]*`, 'g');
+    const noDelimiters = value.replace(delimiterMatcher, '');
+
+    return I18nHelper.formatDecimal(
+      noDelimiters,
+      this.validatePrecision()
+    );
+  }
+
+  validatePrecision = () => {
+    const {
+      precision,
+      maxPrecision
+    } = this.props;
+
+    if (precision > maxPrecision) {
+      Logger.warn(`Precision cannot be greater than ${maxPrecision}`);
+      return maxPrecision;
+    }
+
+    if (!precision || precision < 0) {
```

Comment:
won't `!precision` be truthy if precision is `0`? should it not be `precision === false`? To be honest I'm not entirely clear what scenario this is trying to resolve?

Reply:
`0` is a falsey value in Javascript. With `!precision`, I was trying to account for that `0` but also, the Storybook knob passing an empty `string`.

Reply:
What I mean is, shouldn't we respect `0` as a value? This looks like it will apply the default value if 0 is passed in? Maybe it should be `precision === '' || precision < 0`?

There should probably be a limit though to what values we cater for vs quirks that storybook are throwing at us? I mean PropTypes are there to guide the developer to the correct value type, _should_ we also be coding around invalid types?